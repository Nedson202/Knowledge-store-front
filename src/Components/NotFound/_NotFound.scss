.not-found {
  position: absolute;
  width: 100%;
  max-width: 700px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);

  &__404 {
    font-size: 200px;
    transform: rotate(-5deg);
    color: $grey_3 !important;
  }
}
