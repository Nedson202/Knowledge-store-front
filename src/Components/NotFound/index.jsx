import React from 'react';
import MetaTags from 'react-meta-tags';

const NotFound = () => (
  <div className="not-found text-center">
    <MetaTags>
      <meta
        name="description"
        content="Page is not found"
      />
      <meta property="og:title" content="Page not found" />
    </MetaTags>

    <h1 className="not-found__404">404!</h1>
    <h1 className="not-found__text">
      Sorry this page does not exist.
    </h1>
  </div>
);

export default NotFound;
