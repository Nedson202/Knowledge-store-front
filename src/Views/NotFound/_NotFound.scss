.not-found {
  position: absolute;
  width: 100%;
  max-width: 700px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  font-weight: 100;

  &__404 {
    font-size: 200px;
    transform: rotate(-5deg);
    margin-bottom: 16px;
  }
}
