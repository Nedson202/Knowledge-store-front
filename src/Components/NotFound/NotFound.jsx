import React, { Component } from 'react';
import './_NotFound.scss';

class NotFound extends Component {
  render() {
    return (
      <div className="not-found text-center">
        <h1 className="not-found__404">404!</h1>
        <h1 className="not-found__text">
          Sorry this page seems to have been eaten.
        </h1>
      </div>
    );
  }
}

export default NotFound;
