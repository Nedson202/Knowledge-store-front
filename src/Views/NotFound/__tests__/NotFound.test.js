import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import NotFound from '..';

test('NotFound component should render message', () => {
  const { getByText, asFragment } = render(<NotFound />);

  getByText('404!');
  getByText('Sorry this page does not exist.');

  expect(asFragment()).toMatchSnapshot();
});
