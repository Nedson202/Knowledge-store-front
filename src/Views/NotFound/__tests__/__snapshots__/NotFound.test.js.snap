// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render message 1`] = `
<DocumentFragment>
  <div
    class="not-found text-center"
  >
    <h1
      class="not-found__404"
    >
      404!
    </h1>
    <h1
      class="not-found__text"
    >
      Sorry this page does not exist.
    </h1>
  </div>
</DocumentFragment>
`;
